File: backend/src/models/todo.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Todo {
    pub id: String,
    pub course_id: String,
    pub title: String,
    pub due_date: String,
    pub status: String,
    pub completed_at: Option<String>,
    pub is_archived: bool,
    pub updated_at: String,
    pub sync_state: String,
    pub last_synced_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NewTodoRequest {
    pub course_id: String,
    pub title: String,
    pub due_date: String,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateTodoRequest {
    pub title: Option<String>,
    pub due_date: Option<String>,
    pub status: Option<String>,
}

File: client/Taskion/Taskion/Models/Todo.swift
import Foundation

struct Todo: Codable, Identifiable {
    let id: String
    let courseId: String
    let title: String
    let dueDate: String
    let status: String
    let completedAt: String?
    let isArchived: Bool
    let updatedAt: String
    let syncState: String
    let lastSyncedAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case courseId = "course_id"
        case title
        case dueDate = "due_date"
        case status
        case completedAt = "completed_at"
        case isArchived = "is_archived"
        case updatedAt = "updated_at"
        case syncState = "sync_state"
        case lastSyncedAt = "last_synced_at"
    }
}

struct NewTodoRequest: Codable {
    let courseId: String
    let title: String
    let dueDate: String
    let status: String

    enum CodingKeys: String, CodingKey {
        case courseId = "course_id"
        case title
        case dueDate = "due_date"
        case status
    }
}

struct UpdateTodoRequest: Codable {
    let title: String?
    let dueDate: String?
    let status: String?

    enum CodingKeys: String, CodingKey {
        case title
        case dueDate = "due_date"
        case status
    }
}